_posts/2019-04-06-omakase-relay.markdown

```diff
@@ -0,0 +1,240 @@
+find ourselves defending this decision to new engineers during onboarding and to the public at large.
+
+Which makes this a perfect blog post topic, so let's have a deep dive into what makes Relay compelling for Artsy's
+engineering team.
+
+<!-- more -->
+
+# What problem does Relay solve?
+
+Relay is an API client for GraphQL, it comes in two parts: a compiler and a set of front-end components. Relay aims
+to provide a really tight binding between your GraphQL API and your view hierarchy. When you build data-driven
+apps, Relay removes a whole suite of non-business logic from your application.
+
+Relay handles:
+
+- Data binding (API -> props)
+- Cache management (invalidation, updates etc)
+- Consistent bi-directional pagination abstractions
+- Multiple query consolidation (e.g. consolidate all API requests to one request)
+- Declarative data mutation (describe how data should change, instead of doing it)
+- UI best practices baked in (optimistic response rendering, cheap rollbacks)
+- AOT query generation (allowing you to persist queries)
+
+By taking the responsibilities of the grunt work for most complex apps and moving it into Relay you get
+Facebook-scale best-practices and can build on top of that.
+
+# How does it work?
+
+You write a set of Relay components, you always start with a [`QueryRenderer`][query] then use a tree of either
+[`FragmentContainer`][frag], [`RefetchContainer`][re] or [`PaginationController`][pag]s. You mostly use
+`FragmentContainer`s so I'll focus on that here.
+
+A `FragmentContainer` is based on a [GraphQL fragment][gql-frag]. If you've never used a fragment, they are an
+abstraction that lets you declare shared fields on a specific GraphQL type to reduce duplication up your queries.
+For example:
+
+```
+query GetPopularArtistsAndMyFavs {
+  me {
+    artists {
+      id
+      name
+      bio
+    }
+  }
+  popularArtists {
+    id
+    name
+    bio
+  }
+}
+```
+
+To move this query to use fragments:
+
+```
+query GetPopularArtistsAndMyFavs {
+  me {
+    artists {
+      ...ArtistMetadata
+    }
+  }
+  popularArtists {
+    ...ArtistMetadata
+  }
+}
+
+fragment ArtistMetadata on Artist {
+  id
+  name
+  bio
+}
+```
+
+It's tiny a bit longer, but you have a guarantee that the data is consistent across both sets of artists. Now that
+you have a rough idea of what a GraphQL fragment is, let's look at what a `FragmentContainer` looks like. Here's a
+simplified [profile page] from the Artsy iOS app:
+
+```ts
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { MyProfile_me } from "__generated__/MyProfile_me.graphql"
+
+interface Props extends ViewProperties {
+  me: MyProfile_me
+}
+
+export class MyProfile extends React.Component<Props> {
+  render() {
+    return (
+      <View>
+        <Header>
+          <ProfilePhoto initials={props.me.initials} image={props.me.image} />
```

Comment:
It could, but it is meant to be a simplification of the existing Component, so differing that much (by moving to a func ) probably would detract from the point